src/pages/Home.jsx

```diff
@@ -1,11 +1,36 @@
-import React from "react";
+import React, { useState } from "react";
+import {Country} from "../components/Country"
+import data from '../assets/CountriesData.json'
+
 
 const Home = () => {
+    const [cards, setCards]=useState(data)
+    const onSearch=(e)=>{
+        const newData= data.filter((card)=>card.name.toLowerCase().includes(e.target.value.toLowerCase()));
+        console.log(newData)
+        setCards(newData);
+    
+    }
+    
     return (
-        // TODO: Home page
-        // Render Country component (components/Country.jsx) for each country
-        // Take data from (assets/CountriesData.json)
-        <div>Home</div>
+        <div>
+            <input type="text" className="search-input" placeholder="Search for a country..." onChange={onSearch}></input>
+            <div className="countries-grid"> 
+                { cards.map((card)=>{
+                    return (
+                    <Country 
+                    key={card.name}
+                    name={card.name} 
```

Comment:
You can add another props to you Card component like this : 

```js
export const Country = ({ name, flag, population, region, capital, ...props }) => { 
  return (
    <div {...props}>
      <img src={flag} alt={`${name} flag`} width="100" />
      <h2>{name}</h2>
      <p>Population: {population.toLocaleString()}</p>
      <p>Region: {region}</p>
      <p>Capital: {capital}</p>
    </div>
  );
};

```

Then all you need to do it call you component like this:

```js
<Country
     //Your props goes in here
      {...country}}
>
```